prompt/_generic_list.ts

```diff
@@ -111,31 +111,52 @@ export abstract class GenericList<T, V, S extends GenericListSettings<T, V>>
       this.options = this.settings.options.slice();
     } else {
       this.options = this.settings.options
-        .filter((option: GenericListOptionSettings) =>
-          match(option.name) ||
-          (option.name !== option.value && match(option.value))
-        )
-        .sort((a: GenericListOptionSettings, b: GenericListOptionSettings) =>
-          distance(a.name, input) - distance(b.name, input)
-        );
```

Comment:
All the changes in this file are just pulling code into smaller functions, so I can reuse bits of it in the `Select` module